store/src/java/com/zimbra/cs/service/mail/SaveDraft.java

```diff
@@ -81,7 +87,28 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         boolean wantModSeq = request.getAttributeBool(MailConstants.A_WANT_MODIFIED_SEQUENCE, false);
         Element msgElem = request.getElement(MailConstants.E_MSG);
 
-        int id = (int) msgElem.getAttributeLong(MailConstants.A_ID, Mailbox.ID_AUTO_INCREMENT);
+        /*Added for ZCS-14408*/
+        String mailAddress = null;
+        List<Element> elements = msgElem.listElements(MailConstants.E_EMAIL);
+        for (Element element : elements) {
+            if (element.getAttribute(MailConstants.A_ITEM_TYPE, null).equals(MailConstants.A_FLAGS)) {
+                mailAddress = element.getAttribute(MailConstants.A_ADDRESS, null);
+            }
+        }
+        Provisioning prov = Provisioning.getInstance();
+        Account delegatorAccount;
+        Mailbox delegatorMbox = null;
+        if (mailAddress != null) {
+            delegatorAccount = prov.get(AccountBy.name, mailAddress, zsc.getAuthToken());
+            String mRequestedAccountId = delegatorAccount.getId();
+            delegatorMbox = MailboxManager.getInstance().getMailboxByAccountId(mRequestedAccountId, true);
+        }
+        if (!mbox.getAccountId().equals(delegatorMbox.getAccountId())) {
```

Comment:
delegatorMbox should be checked for null here

Reply:
Delegator Mailbox can't be null because we are fetching it from "From Address" of the Request. Since the "From Address" can't be null, we will always get some value of Delegator Mailbox.